rholang/examples/linking/tests/LinkedList_test.rho

```diff
@@ -0,0 +1,265 @@
+            }
+          }
+        } |
+        contract tail_test(_) = {
+          //return the tail of the list
+          new NonEmptyList in {
+            LinkedList(1, NonEmptyList) | for(tl <- NonEmptyList) {
+              prepend(2, tl, NonEmptyList) | for(list <- NonEmptyList) {
+                new result1, result2, answer1, answer2 in {
+                  tail(list, result1) | tail(emptyList, result2) |
+                  answer1(tl) | answer2(Nil) |
+                  for(r1 <- result1; r2 <- result2; a1 <- answer1; a2 <- answer2) {
+                    match [r1 == a1, r2 == a2] with [true, true] => {
+                      "Tail: Pass".display("\n")
+                    }
+                    _ => { "Tail: Fail".display("\n") }
+                  }
+                }
+              }
+            }
+          }
+        } |
+        contract prepend_test(_) = {
+          //create a new list with the supplied head attached
+          //to the supplied tail
+          new hd, tl, result in {
+            hd(1) | tl(emptyList) |
+            for(h <- hd; l <- tl) {
+              prepend(h, l, result) | for(r <- result) {
+                head(r, hd) | tail(r, tl) | 
+                for(rh <- hd; rl <- tl) {
+                  match [rh == h, rl == l] with [true, true] => {
+                    "Prepend: Pass".display("\n")
+                  }
+                  _ => { "Prepend: Fail".display("\n") }
+                }
+              }
+            }
+          }
+        } |
+        contract fold_test(_) = {
+          //left associative fold
+          new sum, rList, result in {
+            contract sum(head, accVal, return) = {
+              return(head + accVal)
+            } |
+            range(1, 10, rList) | for(list <- rList) {
+              fold(list, 0, sum, result) |
+              for(r <- result) {
+                match r == 45 with true => { "Fold: Pass".display("\n") }
+                _ => { "Fold: Fail".display("\n") }
+              }
+            }
+          }
+        } |
+        contract reverse_test(_) = {
+          //reverse the order of the elements in the list
+          new rangeList, reverseList, verifier in {
+            range(0, 10, rangeList) |  for(list <- rangeList) {
+              reverse(list, reverseList) | for(list2 <- reverseList) {
+                contract verifier(head, accVal, return) = {
+                  match accVal with [i, result] => {
+                    new newResult in {
+                      and(head == (9 - i), result, newResult) |
+                      for(nr <- newResult){ return([i + 1, nr]) }
+                    }
+                  }
+                } |
+                new vResult in {
+                  fold(list2, [0, true], verifier, vResult) |
+                  for(r <- vResult) {
+                    match r with [_, true] => { "Reverse: Pass".display("\n") }
+                    _ => { "Reverse: Fail".display("\n") }
+                  }
```

Comment:
This is quite the test. We really need some shortcut that just allows us to do the equivalent of

`assert list == [9,8,7,6,5,4,3,2,1]`

Reply:
Yeah, I should probably write a little tester library which could be imported to do stuff like that.